sixtools/plotting_functions.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.patches import Rectangle
+from mpl_toolkits.axes_grid1 import make_axes_locatable
+
+
+def plot_frame(ax, frame, light_ROIs=[], cax=None, **kwargs):
+    """
+    Plot the whole frame coming out of the detector and check the ROIs
+
+    Parameters
+    ----------
+    ax : matplotlib axis object
+        axis to plot frame on
+    frame : array
+        2D image to plot
+    light_ROIs : list of [slice, slice]
+        Regions of interest to plot as white rectangles
+    cax : None or matplotlib axis object
+        axis to plot colorbar on
+        If none a colorbar is created
+    **kwargs : dictionary
+        passed onto matplotlib.pyplot.imshow.
+
+    Returns
+    -------
+    art : matplotlib artist
+        artist from the plot
+    cax : matplotlib axis
+        colorbar axis object
+    cb : matplotlib colorbar
+        colorbar
+    """
+
+    defaults = {'vmin': np.nanpercentile(frame, 5),
+                'vmax': np.nanpercentile(frame, 95)}
+
+    kwargs.update({key: val for key, val in defaults.items()
+                   if key not in kwargs})
+
+    art = ax.imshow(frame, **kwargs)
+    if cax is None:
+        divider = make_axes_locatable(ax)
+        cax = divider.append_axes("right", size="2%", pad=0.1)
+
+    cb = plt.colorbar(art, cax=cax)
+
+    for light_ROI in light_ROIs:
+        row_i = light_ROI[0].start
+        row_f = light_ROI[0].stop
+        col_i = light_ROI[1].start
+        col_f = light_ROI[1].stop
+
+        box = Rectangle((col_i, row_i), (col_f-col_i), (row_f-row_i),
+                        facecolor='none', edgecolor='w')
+        ax.add_patch(box)
+
+    cax.set_xticks([])
+    cax.set_yticks([])
+    return art, cax, cb
+
+
+def plot_scan(ax, scan, event_labels=None, xlabel='Energy loss',
```

Comment:
As above, this seems to be a relatively universally useful plot function with nothing that makes it 'NSLS-II' specific is there a reason not to move this up to scikit-beam/rixs. What do you think?

Reply:
I have something of a similar opinion to plot_frame discussed previously. 

This function assumes a 4D array of RIXS spectra with structure event, image_index, y, I. Since this is not the standard scikit-beam/rixs API I prefer keeping it here.